packages/react-reconciler/src/ReactFiberRoot.new.js

```diff
@@ -35,6 +37,12 @@ import {initializeUpdateQueue} from './ReactUpdateQueue.new';
 import {LegacyRoot, ConcurrentRoot} from './ReactRootTags';
 import {createCache, retainCache} from './ReactFiberCacheComponent.new';
 
+export type RootState = {
+  element: any,
+  cache?: Cache | null,
+  transitions?: Transitions | null,
```

Comment:
Let's just make both of these fields required. No `?`, just `Cache | null` and `Transitions | null`. That way the value is never undefined.